rasa/core/featurizers/tracker_featurizers.py

```diff
@@ -178,14 +183,60 @@ def training_states_and_actions(
         Returns:
             A tuple of list of states and list of actions.
         """
+
+        rasa.shared.utils.io.raise_deprecation_warning(
+            "'training_states_and_actions' is being deprecated in favor of "
+            "'training_states_and_labels'."
+        )
+
+        return self.training_states_and_labels(
+            trackers, domain, omit_unset_slots=omit_unset_slots
+        )
+
+    def training_states_and_labels(
+        self,
+        trackers: List[DialogueStateTracker],
+        domain: Domain,
+        omit_unset_slots: bool = False,
+    ) -> Tuple[List[List[State]], List[List[Text]]]:
+        """Transforms list of trackers to lists of states and labels.
+
+        Args:
+            trackers: The trackers to transform
+            domain: The domain
+            omit_unset_slots: If `True` do not include the initial values of slots.
+
+        Returns:
+            A tuple of list of states and list of labels.
+        """
         (
             trackers_as_states,
-            trackers_as_actions,
+            trackers_as_labels,
             _,
-        ) = self.training_states_actions_and_entities(
+        ) = self.training_states_labels_and_entities(
+            trackers, domain, omit_unset_slots=omit_unset_slots
+        )
+        return trackers_as_states, trackers_as_labels
+
+    def training_states_labels_and_entities(
+        self,
+        trackers: List[DialogueStateTracker],
+        domain: Domain,
+        omit_unset_slots: bool = False,
+    ) -> Tuple[List[List[State]], List[List[Text]], List[List[Dict[Text, Any]]]]:
+        """Transforms list of trackers to lists of states, labels, and entity data.
+
+        Args:
+            trackers: The trackers to transform
+            domain: The domain
+            omit_unset_slots: If `True` do not include the initial values of slots.
+
+        Returns:
+            A tuple of list of states, list of labels and list of entity data.
+        """
+        return self.training_states_actions_and_entities(
```

Comment:
If you swap the method logic you can throw out a lot of repetitive in the subclasses:
1. let this method here, `training_states_labels_and_entities`, raise the not implemented error
2. let `training_states_actions_and_entities` print the deprecation warning and call `training_states_labels_and_entities`
3. remove all the implementations for `training_states_actions_and_entities` from subclasses

Reply:
Will do!